frontend/src/api/generate.go

```diff
@@ -0,0 +1,85 @@
+// Package main is used to generate the backend data model as typescript interfaces.
+//
+//go:generate bash generate.sh
```

Comment:
The makefile triggers go generate on this file. Which runs the bash script. Which runs this app.

GitHub Actions and DockerHub both run the makefile so this is fully pipelined.